### PR TITLE
CSRF attack prevention

### PR DESCRIPTION
**Pull request by adamtpowell**
Prevents CSRF attacks allowing account password changing from malicious sites.

For further security improvements, require password authentication for all dangerous admin activities.

### CODE REVIEW

File: template/menu.html [hunk 1]
```diff
@@ -3,7 +3,9 @@
     {{#user}}
       <h3 class="panel-title">Hello {{user.display_name}}</h3>
       <form method='post'>
-        <button type="submit" name='action' value='logout' class="btn btn-default">Log out</button>
+				{{> csrf_token}}
+				<button type="submit" name='action' value='logout' class="btn btn-default">Log out</button>
+				{{> csrf_token}}
```
Comment (liambaloh) [inline template/menu.html]: Two CSRF tokens?

Comment (adamtpowell) [reply template/menu.html]: its twice as safe that way

But actually I accidentally added one before I added it to all the forms.

**Comment by liambaloh** [comment]
Unfortunately I don't know enough about cross site request forgery to perform the requested code review... So if you believe this helps, feel free to merge it into master 👍

**Comment by adamtpowell** [comment]
The way it was before, if a malicious site included the edit user form, any time an admin logged into that site it could use their credentials to change the password of a user (another admin). I performed that on my own account and changed its password.

So pretty much any attacker could very easily get admin access to the site without anybody knowing about it.

File: php/anti-csrf.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+if (empty($_SESSION['csrf_token'])) {
```
[the remark this answers is not in the document]
Comment (adamtpowell) [reply php/anti-csrf.php]: I don't think that's an attack vector (the variable will only be reset with a new session, the value is still 32 bytes of randomness), but it definitely is better style having it separate. I made the change.

**Pull request merged**